package/crds/efs.aws.crossplane.io_mounttargets.yaml

```diff
@@ -0,0 +1,208 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.0
+  creationTimestamp: null
+  name: mounttargets.efs.aws.crossplane.io
+spec:
+  group: efs.aws.crossplane.io
+  names:
+    categories:
+    - crossplane
+    - managed
+    - aws
+    kind: MountTarget
+    listKind: MountTargetList
+    plural: mounttargets
+    singular: mounttarget
+  scope: Cluster
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.conditions[?(@.type=='Ready')].status
+      name: READY
+      type: string
+    - jsonPath: .status.conditions[?(@.type=='Synced')].status
+      name: SYNCED
+      type: string
+    - jsonPath: .metadata.annotations.crossplane\.io/external-name
+      name: EXTERNAL-NAME
+      type: string
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: MountTarget is the Schema for the MountTargets API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: MountTargetSpec defines the desired state of MountTarget
+            properties:
+              deletionPolicy:
+                default: Delete
+                description: DeletionPolicy specifies what will happen to the underlying external when this managed resource is deleted - either "Delete" or "Orphan" the external resource.
+                enum:
+                - Orphan
+                - Delete
+                type: string
+              forProvider:
+                description: MountTargetParameters defines the desired state of MountTarget
+                properties:
+                  fileSystemID:
+                    description: The ID of the file system for which to create the mount target.
+                    type: string
```

Comment:
Can this come from a secretRef? Since the EFS will write it's fileSystemID to a secret with https://doc.crds.dev/github.com/crossplane/provider-aws/efs.aws.crossplane.io/FileSystem/v1alpha1@v0.18.1#spec-writeConnectionSecretToRef

Reply:
can you have look for latest changes ?
i added referencers - thinks this fits 

```
apiVersion: efs.aws.crossplane.io/v1alpha1
kind: MountTarget
metadata:
  name: example
spec:
  forProvider:
    region: us-east-1
    fileSystemIDRef: 
      name: example
    subnetIDRef:
      name: sample-subnet1
    securityGroupsRefs:
      - name: sample-cluster-sg
  providerConfigRef:
    name: example

```

Reply:
I'm not quite sure how that would pull from the secret created by the EFS component but happy to take your word for it. Is that just the secret name? It's quite likely I don't understand how the Ref's work. 
Would love to see an example combined with `kind: FileSystem`
Don't let me block you though - just questions trying to understand how this works.

Reply:
it is working like this:

```
apiVersion: efs.aws.crossplane.io/v1alpha1
kind: FileSystem
metadata:
  name: ${FILESYSTEM_NAME}
spec:
  forProvider:
    region: us-east-1
  providerConfigRef:
    name: default
---
apiVersion: efs.aws.crossplane.io/v1alpha1
kind: MountTarget
metadata:
  name: example
spec:
  forProvider:
    region: us-east-1
    fileSystemIDRef: 
      name: ${FILESYSTEM_NAME}
    subnetIDRef:
      name: sample-subnet1
    securityGroupsRefs:
      - name: sample-cluster-sg
  providerConfigRef:
    name: example
```

Reply:
@haarchri I have a follow-up question.
Assuming that we have provisioned a `FileSystem` and `MountTarget`(s) via Crospslane, how do I use it?
If I use the aws-efs-csi-driver I would have to create a `StorageClass` referencing the _file-system-id_ of the `FileSystem` I created earlier. This class can then be referenced by metadata.name in one of my PVCs.

But I cannot define the_file-system-id_  to which my StorageClass is referring explicitly, Nor can I tell my StorageClass to look it up. Or am I missing something?

Do I have to break the automation here and find out the ID manually?

Reply:
added a public gist in comments to make it more clearer